clippy_lints/src/disallowed_method.rs

```diff
@@ -63,31 +82,30 @@ impl_lint_pass!(DisallowedMethod => [DISALLOWED_METHOD]);
 
 impl<'tcx> LateLintPass<'tcx> for DisallowedMethod {
     fn check_crate(&mut self, cx: &LateContext<'_>, _: &Crate<'_>) {
-        for path in &self.disallowed {
+        for (path, reason) in &self.disallowed {
             let segs = path.iter().map(ToString::to_string).collect::<Vec<_>>();
             if let Res::Def(_, id) = clippy_utils::path_to_res(cx, &segs.iter().map(String::as_str).collect::<Vec<_>>())
             {
-                self.def_ids.insert((id, path.clone()));
+                self.def_ids.insert((id, path.clone(), reason.clone()));
             }
         }
     }
 
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
         if let Some(def_id) = fn_def_id(cx, expr) {
-            if self.def_ids.iter().any(|(id, _)| def_id == *id) {
+            if let Some((_, _, reason)) = self.def_ids.iter().find(|(id, _, _)| def_id == *id) {
                 let func_path = cx.get_def_path(def_id);
                 let func_path_string = func_path
                     .into_iter()
                     .map(Symbol::to_ident_string)
                     .collect::<Vec<_>>()
                     .join("::");
 
-                span_lint(
-                    cx,
-                    DISALLOWED_METHOD,
-                    expr.span,
-                    &format!("use of a disallowed method `{}`", func_path_string),
-                );
+                let msg = match reason {
+                    None => format!("use of a disallowed method `{}`", func_path_string),
+                    Some(reason) => format!("use of a disallowed method `{}`: {}", func_path_string, reason),
+                };
+                span_lint(cx, DISALLOWED_METHOD, expr.span, &msg);
```

Comment:
If a reason is given, this should be displayed in a `note:` in the lint output, not in the error message. You can just use `span_lint_and_note` here to do that.

Reply:
nice, i did that